src/graphql/thegraph/PoolData.ts

```diff
@@ -0,0 +1,64 @@
+import { ApolloClient, InMemoryCache } from '@apollo/client'
+import { ChainId } from '@uniswap/sdk-core'
+import gql from 'graphql-tag'
+import { useMemo } from 'react'
+
+import { usePoolDataQuery } from './__generated__/types-and-hooks'
+import { CHAIN_SUBGRAPH_URL } from './apollo'
+
+gql`
+  query PoolData($poolId: [ID!]) {
+    data: pools(where: { id_in: $poolId }, orderBy: totalValueLockedUSD, orderDirection: desc, subgraphError: allow) {
+      id
+      feeTier
+      liquidity
+      sqrtPrice
+      tick
+      token0 {
+        id
+        symbol
+        name
+        decimals
+        derivedETH
+      }
+      token1 {
+        id
+        symbol
+        name
+        decimals
+        derivedETH
+      }
+      token0Price
+      token1Price
+      volumeUSD
+      volumeToken0
+      volumeToken1
+      txCount
+      totalValueLockedToken0
+      totalValueLockedToken1
+      totalValueLockedUSD
+    }
+    bundles(where: { id: "1" }) {
+      ethPriceUSD
+    }
+  }
+`
+
+export function usePoolData(poolAddress: string, chainId?: ChainId) {
+  const poolId = [poolAddress]
+  const apolloClient = useMemo(
+    () =>
+      new ApolloClient({
+        cache: new InMemoryCache(),
+        uri: CHAIN_SUBGRAPH_URL[chainId ?? ChainId.MAINNET],
+      }),
+    [chainId]
+  )
```

Comment:
should the client be instantiated elsewhere and just consumed inside the hook?

Reply:
Data for each chain comes from its own subgraph which requires its own apollo client. We only know the client to use at pageload as users can query pools across chains separate from their connected chain so I placed it here at the query level. We could in theory instantiate it on the page and then pass the client to the query?

Reply:
I was imagining we instantiate all the apollo clients on page load, then select the appropriate one in the hook when we need to fetch data.

Reply:
My intuition is that client instantiation is somewhat expensive and we wouldn't want to create/dispose for each data request